src/utils/visas.js

```diff
@@ -1,44 +1,51 @@
+import { formatToDateObject, regexDate } from "./date";
+
 import { useProjects } from "@/state/projects";
+import { fullName } from "@/utils/users";
+
 const { fetchFolderProjectUsers } = useProjects();
 
-const getUnmatchedUsers = async (visaList, baseInfo) => {
-  let allProjectsUsers = new Set();
+const isDateConform = date => {
+  const dateToCompare = formatToDateObject(date);
+  const today = new Date();
+  today.setHours(0, 0, 0, 0);
 
-  visaList.forEach(({ document }) => allProjectsUsers.add(document.parentId));
-  allProjectsUsers = Array.from(allProjectsUsers);
+  return date.match(regexDate) && dateToCompare.getTime() >= today.getTime();
+};
 
-  const usersByProjects = await Promise.all(
-    allProjectsUsers.map(async folderId => ({
-      folderId,
-      users: await fetchFolderProjectUsers(
-        {
-          id: baseInfo.projectPk,
-          cloud: {
-            id: baseInfo.cloudPk
-          }
-        },
-        { id: folderId }
-      )
-    }))
+const getUserList = async ({ baseInfo, fileParentId }, validationList) => {
```

Comment:
This function should belong to the visa's state.
Utility functions should be independent from the API / Services.